### ISSUE TITLE
Can not match against an attribute that returns an enumerable

### ISSUE THREAD

**Issue by IanWhitney** [umn-asr/query_string_search#7]
A somewhat contrived example:

``` ruby
class Movie
  def home_release_formats
    ["DVD", "BD", "HULU"]
  end
end
```

`home_release_formats` can not be used as a query-string filter. 

`movies?q=home_release_formats=DVD` or `movies?q=home_release_formats=DVD|BD` should both work. Currently they do not.

**Comment by IanWhitney** [comment]
Relevant problem code is 

https://github.com/umn-asr/query_string_search/blob/master/lib/query_string_search/matchers/match_attribute_value.rb#L4

and 

https://github.com/umn-asr/query_string_search/blob/master/lib/query_string_search/matchers/match_multiple_attribute_values.rb#L4

The matchers always convert the return value of `home_release_formats` to a string. So `["DVD", "BD", "HULU"]` becomes `"[\"DVD\", \"BD\", \"HULU\"]"`

For a single attribute match, that string is then compared to the query string value, which obviously won't match. And for the multi-attribute match, it tries to find that string in an array, which also won't match.

**Comment by IanWhitney** [comment]
This is done in the branch https://github.com/umn-asr/query_string_search/tree/iw/feature/support_attributes_that_return_sets

Since the feature relies on code in #10, I'll rebase and do a new PR after #10 is merged.

**Comment by IanWhitney** [comment]
Implemented in #12
